### PR TITLE
doc/dev: add "Deploy a cluster for manual testing" section

### PR DESCRIPTION
**Pull request by smithfarm**
http://tracker.ceph.com/issues/15218 Fixes: #15218

Signed-off-by: Nathan Cutler <ncutler@suse.com>

### CODE REVIEW

File: doc/dev/index.rst [hunk 1]
```diff
@@ -1119,6 +1119,53 @@ there::
...
+
+This builds packages from the Ceph git repository and branch specified in
+the ``--ceph-git-url`` and ``--ceph`` options, respectively, provisions VMs
+in OpenStack, installs the packages, and deploys a Ceph cluster on them.
+Then, instead of running automated tests, it stops and enters a wait loop.
+
+At this point, the engineer (user) can ssh into the cluster and do whatever
+manual testing is required. This is done by first sshing into the
+teuthology node, as described above, and then sshing to one of the cluster
+nodes. The IP addresses of the cluster nodes can be found in the OpenStack
```
Comment (dachary) [inline doc/dev/index.rst]: I would suggest a simpler approach: "The IP addresses of the cluster nodes can be displayed by running 'openstack server list' on the teuthology machine". The OpenStack credentials and the OpenStack cli are installed there which is convenient.

File: doc/dev/index.rst [hunk 2]
```diff
@@ -1119,6 +1119,53 @@ there::
...
+Horizon dashboard. The basic VMs (or "instances" in OpenStack terminology)
+created by `ceph-workbench ceph-qa-suite`_ are named as follows:
+
+``teuthology`` - the teuthology node
+
+``packages-repository`` - VM where packages are stored
+
+``ceph-rpm-...` - VM where packages are built
+
+Any others are the cluster nodes.
```
Comment (dachary) [inline doc/dev/index.rst]: The VMs named target* are used by tests. When looking at the teuthology log for the test that you want to debug, the **targets** section of the yaml file at the beginning lists the machines that are assigned to the test.

Comment (smithfarm) [inline doc/dev/index.rst]: @dachary I'm looking at the teuthology.log of http://149.202.190.143:8081/ubuntu-2016-03-20_10:38:44-ceph-disk-wip-14984---basic-openstack/ (the test mentioned in this section) and I cannot find a **targets** section. Do you mean the **roles** and **openstack** sections?

Comment (smithfarm) [inline doc/dev/index.rst]: Ah! You must have meant this:

```
2016-03-20T11:39:06.166 INFO:teuthology.task.internal:Locked targets:
  target149202171058.teuthology: null
  target149202171059.teuthology: null
```

**Comment by smithfarm** [comment]
@dachary: Revised and repushed.

**Comment by dachary** [comment]
Indeed that's what I meant and it does not show in the yaml section of the log, my bad.

File: doc/dev/index.rst [hunk 3]
```diff
@@ -1119,6 +1119,70 @@ there::
...
+Then, instead of running automated tests, it stops and enters a wait loop.
+
+The VMs (or "instances" in OpenStack terminology) created by
+`ceph-workbench ceph-qa-suite`_ are named as follows:
+
+``teuthology`` - the teuthology machine
+
+``packages-repository`` - VM where packages are stored
+
+``ceph-*`` - VM where packages are built
```
Comment (smithfarm) [inline doc/dev/index.rst]: EDIT: s/rpm-//

File: doc/dev/index.rst [hunk 4]
```diff
@@ -1119,6 +1119,70 @@ there::
...
+versatile tools that automatically provision Ceph clusters in the cloud and
+run various tests on them in an automated fashion. This enables a single
+engineer, in a matter of hours, to perform thousands of tests that would
+keep dozens of human testers occupied for days or weeks if conducted
+manually.
+
+However, there are times when the automated tests do not cover a particular
+scenario and manual testing is desired. It turns out that it is simple to
+adapt a test to stop and wait after the Ceph installation phase, and the
+engineer can then ssh into the running cluster.
```
Comment (smithfarm) [inline doc/dev/index.rst]: [EDIT] revised the last sentence

File: doc/dev/index.rst [hunk 5]
```diff
@@ -1119,6 +1119,64 @@ there::
...
+The IP addresses of the target machines can be found by running
+``openstack server list`` on the teuthology machine.
+
+The whole process, which takes some time to complete, can be monitored as
+described in `Run the dummy suite`_. Be patient.
+
+Once the target machines are up and running and the test enters its wait
+loop, the engineer can ssh into the target machines and do whatever manual
+testing is required. Use the teuthology machine as jump host.
+
```
Comment (smithfarm) [inline doc/dev/index.rst]: omitted beginning of next chapter - will put it in a new PR

**Pull request merged**